src/display/general/init.go

```diff
@@ -81,30 +88,93 @@ func NewCPUPage(numCores int) *CPUPage {
 
 // InitGeneral initializes all ui elements for the ui rendered by the grofer command
 func (page *MainPage) InitGeneral(numCores int) {
-
 	// Initialize Bar Graph for Memory Chart
-	page.MemoryChart.Title = " Memory (RAM) "
-	page.MemoryChart.TitleStyle = ui.NewStyle(ui.ColorClear)
-	page.MemoryChart.Labels = []string{"Total", "Available", "Used", "Free"}
-	page.MemoryChart.BarWidth = 8
-	page.MemoryChart.BarGap = 9
-	page.MemoryChart.BarColors = []ui.Color{ui.ColorCyan, ui.ColorGreen}
-	page.MemoryChart.LabelStyles = []ui.Style{ui.NewStyle(ui.ColorClear)}
-	page.MemoryChart.NumStyles = []ui.Style{ui.NewStyle(ui.ColorBlack)}
+	page.memoryChartWidget()
+	// Initialize Table for Disk Chart
+	page.diskChartWidget()
+	// Initialize Plot for Network Chart
+	page.networkChartWidget()
+	// Initialize Graph for CPU Usage
+	page.avgCpuGraphWidget()
+	if numCores > 8 || cpuTableVisible {
+		page.cpuTableWidget(numCores)
+	} else {
+		page.cpuGaugeWidget(numCores)
+	}
+	// Initialize Graph for Temperature Table
+	page.temperatureTableWidget()
+	// Set page grid
+	page.setPageGrid(numCores)
+
+}
+
+func (page *MainPage) setPageGrid(numCores int) {
+	w, h := ui.TerminalDimensions()
+	if numCores > 8 || cpuTableVisible {
+		page.Grid.Set(
+			ui.NewCol(
+				0.4,
+				ui.NewRow(0.5, page.AvgCPUGraph),
+				ui.NewRow(0.5, page.CPUTable)),
+			ui.NewCol(
+				0.6,
+				ui.NewRow(0.4, page.MemoryChart),
+				ui.NewRow(0.3, ui.NewCol(0.5, page.NetworkChart), ui.NewCol(0.5, page.TemperatureTable)),
+				ui.NewRow(0.3, page.DiskChart)),
+		)
+	} else {
+		page.Grid.Set(
+			ui.NewCol(
+				0.4,
+				ui.NewRow(0.5, page.AvgCPUGraph),
+				ui.NewRow(0.5, page.CPUGauge)),
+			ui.NewCol(
+				0.6,
+				ui.NewRow(0.4, page.MemoryChart),
+				ui.NewRow(0.3, ui.NewCol(0.5, page.NetworkChart), ui.NewCol(0.5, page.TemperatureTable)),
+				ui.NewRow(0.3, page.DiskChart)),
+		)
```

Comment:
you missed a small thing here, sorry :p 
```suggestion
			ui.NewCol(
				0.4,
				ui.NewRow(0.5, page.AvgCPUGraph),
				ui.NewRow(0.5, page.CPUTable),
			),
			ui.NewCol(
				0.6,
				ui.NewRow(0.4, page.MemoryChart),
				ui.NewRow(0.3, ui.NewCol(0.5, page.NetworkChart), ui.NewCol(0.5, page.TemperatureTable)),
				ui.NewRow(0.3, page.DiskChart),
			),
		)
	} else {
		page.Grid.Set(
			ui.NewCol(
				0.4,
				ui.NewRow(0.5, page.AvgCPUGraph),
				ui.NewRow(0.5, page.CPUGauge),
			),
			ui.NewCol(
				0.6,
				ui.NewRow(0.4, page.MemoryChart),
				ui.NewRow(0.3, ui.NewCol(0.5, page.NetworkChart), ui.NewCol(0.5, page.TemperatureTable)),
				ui.NewRow(0.3, page.DiskChart),
			),
		)
```